framework/src/multiapps/MultiApp.C

```diff
@@ -1108,8 +1108,31 @@ MultiApp::createApp(unsigned int i, Real start_time)
   const auto & input_file = _input_files[input_index];
 
   // create new parser tree for the application and parse
-  app_params.set<std::shared_ptr<Parser>>("_parser") = std::make_shared<Parser>(input_file);
+  auto parser = std::make_unique<Parser>(input_file);
 
+  if (input_file.size())
+  {
+    parser->parse();
+    auto app_type = parser->getAppType();
+    if (app_type.empty() && _app_type.empty())
+      mooseWarning("The application type is not specified for ",
+                   full_name,
+                   ". Please use [Application] block to specify the application type.");
+    if (!app_type.empty() && app_type != _app_type)
+      mooseError("In the ",
+                 full_name,
+                 ", '",
+                 app_type,
+                 "' is not a registered application. The registered application is named: '",
```

Comment:
Where do you check that it's not registered?

Reply:
_app_type is check in MultiApp::createApps(), but it may worth checking app_type as well